src/pages/FoodServings.jsx

```diff
@@ -0,0 +1,195 @@
+import { useEffect, useState } from 'react';
+import { useGet, useMutate } from 'restful-react';
+
+import {
+  Box,
+  Flex,
+  Heading,
+  Radio,
+  RadioGroup,
+  Stack,
+  Text,
+  Button,
+  Alert,
+  AlertIcon,
+  AlertTitle,
+  AlertDescription,
+  CloseButton,
+} from '@chakra-ui/react';
+
+export default function FoodServings() {
+  const [selectedDay, setSelectedDay] = useState('');
+  const [selectedMeal, setSelectedMeal] = useState('');
+  const [nowServing, setNowServing] = useState('');
+  const [eachDayFood, setEachDayFood] = useState([]);
+  const [requestedFoodId, setRequestedFoodId] = useState('');
+  const [alert, setAlert] = useState({ isVisible: false, message: '' });
+
+  const { data: allFoodData } = useGet({
+    path: '/api/admin/food',
+    verb: 'GET',
+  });
+
+  const { mutate: servingSwitch } = useMutate({
+    path: `/api/admin/food/${requestedFoodId}`,
+    verb: 'PATCH',
+  });
+
+  const groupFoodByDay = (foodData) => {
+    const dayForFood = {};
+    foodData.forEach((item) => {
+      if (dayForFood[item.day]) {
+        dayForFood[item.day].push(item);
+      } else {
+        dayForFood[item.day] = [item];
+      }
+    });
+    return dayForFood;
+  };
+
+  useEffect(() => {
+    if (allFoodData) {
+      const { allFood, currentMeal } = allFoodData;
+      setEachDayFood(groupFoodByDay(allFood));
+      const meal = allFood.filter((food) => food.id === currentMeal[0]);
+      if (meal.length === 0) {
+        setNowServing('No meal currently being served!');
+        return;
+      }
+      const mealString = `Now Serving: Day ${meal[0].day} 
+      ${meal[0].name.charAt(0).toUpperCase() + meal[0].name.slice(1)}`;
+      setNowServing(mealString);
+    }
+  }, [allFoodData]);
+
+  const handleDayChange = (day) => {
+    setSelectedDay(day);
+  };
+
+  const handleMealChange = (meal) => {
+    setSelectedMeal(meal);
+    const data = allFoodData.allFood.filter(
+      (food) => food.day === parseInt(selectedDay, 10) && food.name === meal
+    );
+    setRequestedFoodId(data[0].id);
+  };
+
+  const handleSubmit = async () => {
+    if (selectedDay && selectedMeal) {
+      try {
+        // Update the backend
+        await servingSwitch();
+
+        setAlert({
```

Comment:
found a bug here, not sure exactly what's going on but I am able to select a meal for a day that doesn't exist.

repo steps:
1. click on day 2 and select breakfast
2. click on day 1
3. click submit

I think what would fix this is instead of saving the selected meal, is saving the id of the meal that's being selected, and then use that id to render the current meal at the top
![image](https://github.com/user-attachments/assets/58fc89c4-462e-4c3a-9592-8df10b4d81a6)